pkgs/games/keeperrl/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv, fetchFromGitHub, makeWrapper, requireFile
+, openal, curl, libogg, libvorbis
+, SDL2, SDL2_image, zlib
+, unfree_assets ? false }:
+
+with stdenv.lib;
+
+let
+  pname = "keeperrl";
+
+in stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  version = "alpha28";
+
+  free-src = fetchFromGitHub {
+    owner = "miki151";
+    repo = "keeperrl";
+    rev = version;
+    sha256 = "0isj8ijn5a89m2r5cxk4lcsq0cydx7c0h87vgr8v5cndm3rd27cy";
+  };
+
+  assets = if unfree_assets then requireFile rec {
+    name = "keeperrl_data_${version}.tar.gz";
+    message = ''
+      This nix expression requires that the KeeperRL art assets are already
+      part of the store. These can be obtained from a purchased copy of the game
+      and found in the "data" directory. Make a tar archive of this directory
+      with
+
+      "tar czf ${name} data"
+
+      Then add this archive to the nix store with
+
+      "nix-prefetch-url file://\$PWD/${name}".
+    '';
+    sha256 = "0115pxdzdyma2vicxgr0j21pp82gxdyrlj090s8ihp0b50f0nk53";
+  } else null;
+
+  sourceRoot = "source";
+
+  srcs = [free-src] ++ lists.optional unfree_assets assets;
```

Comment:
Small style point: it would be nice if the spacing around brackets was consistent throughout. So here should be `[ free-src ]`